Build error of igx-date-picker's locale in "strictTemplates" mode

## Description  

When enabling strictTemplates in `tsconfig.json` and setting locale of an `igx-date-picker` to something different than 'en', the build fails with TS2322.

```json
{
  "angularCompilerOptions": {
    "strictTemplates": true
  }
}
```

 * igniteui-angular version: 11.1.8
 * browser: N/A

## Steps to reproduce  

1. Set strictTemplates to true
2. Create a igx-date-picker and set a locale to something different than 'en'. For example, from the docs, https://www.infragistics.com/products/ignite-ui-angular/angular/components/date-picker#internationalization:

```html
<igx-date-picker locale="ja-JP" [value]="date">
  <ng-template igxCalendarHeader let-format>
    {{ format.month.combined | titlecase }}{{format.day.combined }}{{ format.weekday.combined }}
  </ng-template>
  <ng-template igxCalendarSubheader let-format>
    <span (click)="format.yearView()">{{ format.year.combined }}</span>
    <span (click)="format.monthView()">{{ format.month.combined | titlecase }}</span>
  </ng-template>
</igx-date-picker>
```

3. Compile

## Result  

Build error: `error TS2322: Type '"ja-JP"' is not assignable to type '"en"'`

## Expected result  

No build error.

Note: Setting strictTemplates to false "fixes" the error, since the wrong type is ignored.

--- a/projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts
+++ b/projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts
@@ -219,7 +219,7 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
      * <igx-date-picker locale="ja-JP" [value]="date"></igx-date-picker>
      * ```
      */
-    @Input() public locale: 'en';
+    @Input() public locale = 'en';
 
     /**
      * Gets/Sets the default template editor's tabindex.

